TwitchPlaysAssembly/Src/ComponentSolvers/Modded/Misc/ParliamentComponentSolver.cs

```diff
@@ -0,0 +1,494 @@
+								case BillMiddle.rights:
+									letterPosition += 4;
+									break;
+							}
+							break;
+						case Party.liberal:
+							switch (billMiddle)
+							{
+								case BillMiddle.healthcare:
+									letterPosition += 8;
+									break;
+								case BillMiddle.support:
+									letterPosition += 7;
+									break;
+								case BillMiddle.hats:
+									letterPosition += 6;
+									break;
+								case BillMiddle.freedom:
+									letterPosition -= 7;
+									break;
+								case BillMiddle.vaccines:
+									letterPosition += 2;
+									break;
+								case BillMiddle.rights:
+									letterPosition += 1;
+									break;
+							}
+							break;
+						case Party.socialist:
+							switch (billMiddle)
+							{
+								case BillMiddle.healthcare:
+									letterPosition += 1;
+									break;
+								case BillMiddle.support:
+									letterPosition += 2;
+									break;
+								case BillMiddle.hats:
+									letterPosition += 1;
+									break;
+								case BillMiddle.freedom:
+									break;
+								case BillMiddle.vaccines:
+									letterPosition -= 1;
+									break;
+								case BillMiddle.rights:
+									letterPosition += 4;
+									break;
+							}
+							break;
+						case Party.communist:
+							switch (billMiddle)
+							{
+								case BillMiddle.healthcare:
+									letterPosition += 3;
+									break;
+								case BillMiddle.support:
+									letterPosition += 4;
+									break;
+								case BillMiddle.hats:
+									letterPosition += 3;
+									break;
+								case BillMiddle.freedom:
+									letterPosition -= 3;
+									break;
+								case BillMiddle.vaccines:
+									letterPosition -= 2;
+									break;
+								case BillMiddle.rights:
+									letterPosition += 3;
+									break;
+							}
+							break;
+						case Party.birthday:
+							switch (billMiddle)
+							{
+								case BillMiddle.healthcare:
+									letterPosition += 5;
+									break;
+								case BillMiddle.support:
+									letterPosition += 6;
+									break;
+								case BillMiddle.hats:
+									letterPosition += 6;
+									break;
+								case BillMiddle.freedom:
+									letterPosition -= 2;
+									break;
+								case BillMiddle.vaccines:
+									letterPosition -= 1;
+									break;
+								case BillMiddle.rights:
+									letterPosition += 2;
+									break;
+							}
+							break;
+						case Party.lan:
+							switch (billMiddle)
+							{
+								case BillMiddle.healthcare:
+									letterPosition += 7;
+									break;
+								case BillMiddle.support:
+									letterPosition += 8;
+									break;
+								case BillMiddle.hats:
+									letterPosition += 4;
+									break;
+								case BillMiddle.freedom:
+									letterPosition -= 4;
+									break;
+								case BillMiddle.vaccines:
+									letterPosition -= 2;
+									break;
+								case BillMiddle.rights:
+									letterPosition += 1;
+									break;
+							}
+							break;
+					}
```

Comment:
I fear no code, but that *thing...* it scares me.
```suggestion		
					Dictionary<Party, int[]> positions = new Dictionary<Party, int[]>()
					{
						{ Party.republican,   new[] { 2, 1, 0, -1, -3, 5 } },
						{ Party.democratic,   new[] { 4, 3, 2, -3,  3, 0 } },
						{ Party.conservative, new[] { 6, 5, 4, -5, -2, 4 } },
						{ Party.liberal,      new[] { 8, 7, 6, -7,  2, 1 } },
						{ Party.socialist,    new[] { 1, 2, 1,  0, -1, 4 } },
						{ Party.communist,    new[] { 3, 4, 3, -3, -2, 3 } },
						{ Party.birthday,     new[] { 5, 6, 6, -2, -1, 2 } },
						{ Party.lan,          new[] { 7, 8, 4, -4, -2, 1 } },
					};
					
					int letterPosition = (int) billOpener * 5 + billEnding;
					letterPosition += positions[party, billMiddle];
```